### PR TITLE
feat: cyclic example

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Summary 

This example aims to explain how to implement a custom strategy, taking the MNIST dataset and choosing the cyclic strategy as the one to test.

To run the example, `git switch chore/update-identifiers` in **substrafl** as the feature as not been merged yet.

## Companion PR:

Blocked by https://github.com/Substra/substrafl/pull/142

closes FL-632

### CODE REVIEW

**Review by SdgJlbl — COMMENTED** [review]
Great job, thank you. 
Committing the first batch of comments before I forget, will continue later.

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (SdgJlbl) [inline substrafl_examples/go_further/run_mnist_cyclic.py:408]:
```suggestion
            clean_models (bool): Boolean to indicate if we want to keep intermediate shared states.
```

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (SdgJlbl) [inline substrafl_examples/go_further/run_mnist_cyclic.py:438]:
```suggestion
        """This method is called at each round to perform a series of tasks. For the cyclic
```

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (SdgJlbl) [inline substrafl_examples/go_further/run_mnist_cyclic.py:448]:
```suggestion
            clean_models (bool): Boolean to indicate if we want to keep intermediate shared states.
```

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (SdgJlbl) [inline substrafl_examples/go_further/run_mnist_cyclic.py:449]:
```suggestion
                Only taken into account in ``remote`` mode.
```

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (SdgJlbl) [inline substrafl_examples/go_further/run_mnist_cyclic.py:459]: there might be a smarter way of doing that using `itertools.cycle`, but it's not very important

Comment (SdgJlbl) [reply substrafl_examples/go_further/run_mnist_cyclic.py:459]: https://docs.python.org/3.9/library/itertools.html#itertools.cycle

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (SdgJlbl) [inline substrafl_examples/go_further/run_mnist_cyclic.py:584]:
```suggestion
        """This method decorated with ``@remote_data`` is a method that is executed inside
```

**Review by SdgJlbl — COMMENTED** [review]
Great job, thanks a lot for your work!
I know that it's really not trivial, especially as we are in the process of changing the scope of responsibility between `Strategy` and `Algo`, but one thing that I find that could be made clearer is how the responsibility is shared between these two objects. I think it would help users understand where they should put what part of their code. 
It might also be a good idea to get a review from someone from R&D, as they are the intended users of such documentation, and see if they found some things are missing.

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (ThibaultFy) [reply substrafl_examples/go_further/run_mnist_cyclic.py:459]: Mmmmh it's not that straightforward :/

**Comment by ThibaultFy** [comment]
> would

Thanks a lot for the review.
Fully agree that its the unclear part. But the more I try to explain it, the more it's becoming obvious that it's the wrong abstraction... I added a few lines after the warning section, just before the instantiation of objects. Let me know what you think about it...

**Review by SdgJlbl — APPROVED** [review]
It seems clear to me, but I might be in too deep to be the best person to judge. 

I think it would be valuable to have feedbacks from someone from ML, it can be after the merge if you want to merge before the end of the week, we can always amend the example later.

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (RomainGoussault) [inline substrafl_examples/go_further/run_mnist_cyclic.py:12]: Actually it might be worth asking FunML if this is the correct name

File: substrafl_examples/go_further/run_mnist_cyclic.py
Comment (RomainGoussault) [inline substrafl_examples/go_further/run_mnist_cyclic.py:523]: We can not use the FedAvg Algo because we don't apply delta to the model here?

**Pull request merged**
